### PR TITLE
feat: add basic gh workflow

### PR DESCRIPTION
**Pull request by BobyMCbobs**
replicate the travis jobs

## Proposed Changes

Resolves #.

Changes proposed in this pull request:

- use GitHub Actions

## Production Changes

The following production changes are required to deploy these changes:

- None

## Review

Check the box that applies to this code review.  If necessary please seek help with adding a checklist guide for the reviewer.
When assigning the code review please consider the expertise needed to review the changes.

- [ ] This is a content (documentation, web page etc) only change.
- [x] This is a minor change (meta data, bug fix, improve test coverage etc).
- [ ] This is a larger change (new feature, significant refactoring etc).  Please use the code review guidelines to add a checklist below to guide the code reviewer.

### CODE REVIEW

**Comment by BobyMCbobs** [comment]
All green and passing!

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  golangci-lint:
+    if: ${{ contains(fromJSON('["workflow_call", "push", "pull_request"]'), github.event_name) && startsWith(github.repository, 'GeoNet/') != false }}
+    uses: GeoNet/Actions/.github/workflows/reusable-golangci-lint.yml@main
+  go-vet:
+    if: ${{ contains(fromJSON('["workflow_call", "push", "pull_request"]'), github.event_name) && startsWith(github.repository, 'GeoNet/') != false }}
+    uses: GeoNet/Actions/.github/workflows/reusable-go-vet.yml@main
+  go-test:
+    runs-on: ubuntu-latest
+    env:
+      AWS_REGION: ap-southeast-2
```
Comment (wilsonjord) [inline .github/workflows/build.yml:30]: note: shouldn't be necessary, as tests are written to set their own environment.

**Review by wilsonjord — APPROVED** [review]
Good stuff. Seems to be an equivalent pipeline to current Travis job.

File: .github/workflows/build.yml (hunk 1 above)
Comment (BobyMCbobs) [reply .github/workflows/build.yml:30]: @wilsonjord, ya reckon I should remove it?

Comment (wilsonjord) [reply .github/workflows/build.yml:30]: Originally didn't care, just noting. But on second thoughts, better to remove if possible, if only for the principle that it doesn't add anything of value.

Comment (BobyMCbobs) [reply .github/workflows/build.yml]: Removed in 7dab317

**Review by wilsonjord — APPROVED** [review]

**Comment by BobyMCbobs** [comment]
@wilsonjord, did a rebase and force push.
Seems to be a minor error in the test now
https://github.com/GeoNet/kit/actions/runs/5872096431/job/15922983785?pr=141#step:5:131

**Comment by wilsonjord** [comment]
> @wilsonjord, did a rebase and force push. Seems to be a minor error in the test now https://github.com/GeoNet/kit/actions/runs/5872096431/job/15922983785?pr=141#step:5:131

Interesting...unable to replicate locally, and the Travis build passes ok. The test in question:

```golang
func TestSQSSendWithDelay(t *testing.T) {
	// ARRANGE
	setup()
	defer teardown()

	client, err := New()
	require.Nil(t, err, fmt.Sprintf("Error creating sqs client: %v", err))

	// ACTION
	err = client.SendWithDelay(awsCmdQueueURL(), testMessage, 4) // delay seconds
	time.Sleep(2 * time.Second)
	messageCountBeforeDelay := awsCmdQueueCount()
	time.Sleep(3 * time.Second)
	messageCountAfterDelay := awsCmdQueueCount()

	// ASSERT
	assert.Nil(t, err)
	assert.Equal(t, 0, messageCountBeforeDelay)
	assert.Equal(t, 1, messageCountAfterDelay)
}
```

I think it's a matter of getting the wait times appropriate, maybe the `localstack` / client connection is a bit slower than in Travis/locally, so I need to account for that...anyway, I'm looking into it.

**Comment by BobyMCbobs** [comment]
@wilsonjord, delays end up being flakey. I might suggest using a method like

```go
ready := false
for !ready {
  ready := checkForReady()
  if !ready {
    continue
  }
  ready = true
}
```

**Comment by wilsonjord** [comment]
> @wilsonjord, delays end up being flakey. I might suggest using a method like
> 
> ```go
> ready := false
> for !ready {
>   ready := checkForReady()
>   if !ready {
>     continue
>   }
>   ready = true
> }
> ```

Way ahead of you 😉 https://github.com/GeoNet/kit/pull/147

**Comment by wilsonjord** [comment]
> Seems to be a minor error in the test now
this should resolve, and have it consistently passing https://github.com/GeoNet/kit/pull/147
I

**Pull request merged**